plugins/mergeStyles.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { querySelectorAll, closestByName } = require('../lib/xast.js');
+const { getCssStr, setCssStr } = require('../lib/css-tools');
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'merge multiple style elements into one';
+
+/**
+ * Merge multiple style elements into one.
+ *
+ * @param {Object} document document element
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document) {
+  // collect <style/>s with valid type attribute (preserve order)
+  const styleElements = querySelectorAll(document, 'style');
+
+  // no <styles/>s, nothing to do
+  if (styleElements.length <= 1) {
+    return document;
+  }
+
+  let styles = [];
+  for (let styleElement of styleElements) {
+    if (
+      styleElement.attributes.type &&
+      styleElement.attributes.type !== 'text/css'
+    ) {
+      // skip <style> with invalid type attribute
+      continue;
+    }
+
+    if (closestByName(styleElement, 'foreignObject')) {
+      // skip <foreignObject> content
+      continue;
+    }
+
+    const cssString = getCssStr(styleElement);
+
+    styles.push({
+      styleElement: styleElement,
+
+      mq: styleElement.attributes.media,
+      cssStr: cssString,
+    });
+  }
+
+  let collectedStyles = [];
+  for (let styleNo = 0; styleNo < styles.length; styleNo += 1) {
+    const style = styles[styleNo];
+
+    if (style.mq) {
+      const wrappedStyles =
+        '@media ' + style.mq + ' {' + '\n' + style.cssStr + '\n' + '}';
```

Comment:
use template literal

Reply:
Done (continues in https://github.com/svg/svgo/pull/1381#discussion_r602719351).